diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildSteps.java b/jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildSteps.java
index f5e414f4..cfcea64d 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildSteps.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildSteps.java
@@ -170,7 +170,7 @@ public class BuildSteps {
       Caches.Initializer cachesInitializer,
       String startupMessage,
       String successMessage,
-      StepsRunnerConsumer stepsRunnerConsumer) {
+      BuildSteps.StepsRunnerConsumer stepsRunnerConsumer) {
     this.description = description;
     this.buildConfiguration = buildConfiguration;
     this.cachesInitializer = cachesInitializer;
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerMetadata.java b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerMetadata.java
index 5350b631..d7f1c0e3 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerMetadata.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/LayerMetadata.java
@@ -73,7 +73,7 @@ class LayerMetadata {
   /** The last time the layer was constructed. */
   private final FileTime lastModifiedTime;
 
-  LayerMetadata(ImmutableList<LayerMetadataEntry> entries, FileTime lastModifiedTime) {
+  LayerMetadata(ImmutableList<LayerMetadata.LayerMetadataEntry> entries, FileTime lastModifiedTime) {
     this.entries = entries;
     this.lastModifiedTime = lastModifiedTime;
   }
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerObjectTemplate.java b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerObjectTemplate.java
index 6f678b18..66bd0834 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerObjectTemplate.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerObjectTemplate.java
@@ -21,12 +21,6 @@ import com.google.cloud.tools.jib.image.Layer;
 import com.google.cloud.tools.jib.json.JsonTemplate;
 import javax.annotation.Nullable;
 
-/**
- * Inner JSON template for storing metadata about a layer in the cache as part of {@link
- * CacheMetadataTemplate}.
- *
- * @see CacheMetadataTemplate for example
- */
 public class CacheMetadataLayerObjectTemplate implements JsonTemplate {
 
   /** The reference to the layer. */
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerPropertiesObjectTemplate.java b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerPropertiesObjectTemplate.java
index 67e1d65f..64b17562 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerPropertiesObjectTemplate.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/json/CacheMetadataLayerPropertiesObjectTemplate.java
@@ -22,10 +22,6 @@ import java.util.Collections;
 import java.util.List;
 import javax.annotation.Nullable;
 
-/**
- * Inner JSON template for extra properties for an application layer, as part of {@link
- * CacheMetadataLayerObjectTemplate}.
- */
 public class CacheMetadataLayerPropertiesObjectTemplate implements JsonTemplate {
 
   /** Represents a pair of source files and extraction path. */
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/Port.java b/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/Port.java
index 88f7f90c..ceeecc29 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/Port.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/Port.java
@@ -56,7 +56,7 @@ public class Port {
   private final int port;
   private final Protocol protocol;
 
-  public Port(int port, Protocol protocol) {
+  public Port(int port, Port.Protocol protocol) {
     this.port = port;
     this.protocol = protocol;
   }
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageToTarballTranslator.java b/jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageToTarballTranslator.java
index 2de8a737..a2b1c269 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageToTarballTranslator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageToTarballTranslator.java
@@ -29,7 +29,6 @@ import java.io.IOException;
 import java.nio.file.Path;
 import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
 
-/** Translates an {@link Image} to a tarball that can be loaded into Docker. */
 public class ImageToTarballTranslator {
 
   /** File name for the container configuration in the tarball. */
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/UserCacheHome.java b/jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/UserCacheHome.java
index 3fd83824..a2082151 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/UserCacheHome.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/UserCacheHome.java
@@ -25,20 +25,6 @@ import java.util.Map;
 import java.util.Properties;
 import java.util.logging.Logger;
 
-/**
- * Obtains an OS-specific user cache directory based on the XDG Base Directory Specification.
- *
- * <p>Specifically, from the specification:
- *
- * <ul>
- *   <li>This directory is defined by the environment variable {@code $XDG_CACHE_HOME}.
- *   <li>If {@code $XDG_CACHE_HOME} is either not set or empty, a default equal to {@code
- *       $HOME/.cache} should be used.
- * </ul>
- *
- * @see <a
- *     href="https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html">https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html</a>
- */
 public class UserCacheHome {
 
   private static final Logger logger = Logger.getLogger(UserCacheHome.class.getName());
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/http/BlobHttpContent.java b/jib-core/src/main/java/com/google/cloud/tools/jib/http/BlobHttpContent.java
index 54be3357..b5409159 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/http/BlobHttpContent.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/http/BlobHttpContent.java
@@ -21,7 +21,6 @@ import com.google.cloud.tools.jib.blob.Blob;
 import java.io.IOException;
 import java.io.OutputStream;
 
-/** {@link Blob}-backed {@link HttpContent}. */
 public class BlobHttpContent implements HttpContent {
 
   private final Blob blob;
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/http/Request.java b/jib-core/src/main/java/com/google/cloud/tools/jib/http/Request.java
index 6c00d8ac..7f06885d 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/http/Request.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/http/Request.java
@@ -105,7 +105,7 @@ public class Request {
     return new Builder();
   }
 
-  private Request(Builder builder) {
+  private Request(Request.Builder builder) {
     this.headers = builder.headers;
     this.body = builder.body;
     this.httpTimeout = builder.httpTimeout;
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageFormat.java b/jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageFormat.java
index ed80a034..9f82f8b6 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageFormat.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageFormat.java
@@ -20,7 +20,6 @@ import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
 import com.google.cloud.tools.jib.image.json.OCIManifestTemplate;
 import com.google.cloud.tools.jib.image.json.V22ManifestTemplate;
 
-/** Enumeration of {@link BuildableManifestTemplate}s. */
 public enum ImageFormat {
   Docker(V22ManifestTemplate.class),
   OCI(OCIManifestTemplate.class);
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/image/UnwrittenLayer.java b/jib-core/src/main/java/com/google/cloud/tools/jib/image/UnwrittenLayer.java
index 40d46ff9..554d1300 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/image/UnwrittenLayer.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/image/UnwrittenLayer.java
@@ -20,10 +20,6 @@ import com.google.cloud.tools.jib.blob.Blob;
 import com.google.cloud.tools.jib.blob.BlobDescriptor;
 import com.google.cloud.tools.jib.cache.CachedLayer;
 
-/**
- * A layer that has not been written out and only has the unwritten content {@link Blob}. Once
- * written, this layer becomes a {@link CachedLayer}.
- */
 public class UnwrittenLayer implements Layer {
 
   private final Blob uncompressedBlob;
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/OCIManifestTemplate.java b/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/OCIManifestTemplate.java
index 806b24f2..b97eb50a 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/OCIManifestTemplate.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/OCIManifestTemplate.java
@@ -70,10 +70,10 @@ public class OCIManifestTemplate implements BuildableManifestTemplate {
   private final String mediaType = MANIFEST_MEDIA_TYPE;
 
   /** The container configuration reference. */
-  @Nullable private ContentDescriptorTemplate config;
+  @Nullable private BuildableManifestTemplate.ContentDescriptorTemplate config;
 
   /** The list of layer references. */
-  private final List<ContentDescriptorTemplate> layers = new ArrayList<>();
+  private final List<BuildableManifestTemplate.ContentDescriptorTemplate> layers = new ArrayList<>();
 
   @Override
   public int getSchemaVersion() {
@@ -87,12 +87,12 @@ public class OCIManifestTemplate implements BuildableManifestTemplate {
 
   @Override
   @Nullable
-  public ContentDescriptorTemplate getContainerConfiguration() {
+  public BuildableManifestTemplate.ContentDescriptorTemplate getContainerConfiguration() {
     return config;
   }
 
   @Override
-  public List<ContentDescriptorTemplate> getLayers() {
+  public List<BuildableManifestTemplate.ContentDescriptorTemplate> getLayers() {
     return Collections.unmodifiableList(layers);
   }
 
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestTemplate.java b/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestTemplate.java
index fc6db4cd..9f2bb1ee 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestTemplate.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestTemplate.java
@@ -72,10 +72,10 @@ public class V22ManifestTemplate implements BuildableManifestTemplate {
   private final String mediaType = MANIFEST_MEDIA_TYPE;
 
   /** The container configuration reference. */
-  @Nullable private ContentDescriptorTemplate config;
+  @Nullable private BuildableManifestTemplate.ContentDescriptorTemplate config;
 
   /** The list of layer references. */
-  private final List<ContentDescriptorTemplate> layers = new ArrayList<>();
+  private final List<BuildableManifestTemplate.ContentDescriptorTemplate> layers = new ArrayList<>();
 
   @Override
   public int getSchemaVersion() {
@@ -89,12 +89,12 @@ public class V22ManifestTemplate implements BuildableManifestTemplate {
 
   @Override
   @Nullable
-  public ContentDescriptorTemplate getContainerConfiguration() {
+  public BuildableManifestTemplate.ContentDescriptorTemplate getContainerConfiguration() {
     return config;
   }
 
   @Override
-  public List<ContentDescriptorTemplate> getLayers() {
+  public List<BuildableManifestTemplate.ContentDescriptorTemplate> getLayers() {
     return Collections.unmodifiableList(layers);
   }
 
diff --git a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java
index 78bdfa50..18aa4153 100644
--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryErrorException.java
@@ -18,10 +18,6 @@ package com.google.cloud.tools.jib.registry;
 
 import javax.annotation.Nullable;
 
-/**
- * Thrown when an HTTP request to a registry endpoint failed with errors as defined in {@link
- * ErrorCodes}.
- */
 public class RegistryErrorException extends RegistryException {
 
   RegistryErrorException(String message, @Nullable Throwable cause) {
